### PR TITLE
[FLINK-36791][table] Remove FlinkPruneEmptyRules.

### PR DESCRIPTION
**Pull request by liuyongvs**
## What is the purpose of the change

A bit reasoning for the change
high calcite version has fixed, current calcite version is 1.34

## Verifying this change

This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by davidradl** [comment]
Reviewed by Chi on 28/11/24 business as usual progressing with committer involved.

**Comment by liuyongvs** [comment]
hi @snuyanzin  will you help review this ?

File: flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PruneEmptyRulesTest.xml
Comment (snuyanzin) [inline flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PruneEmptyRulesTest.xml:1]: which test still uses it?

Comment (liuyongvs) [reply flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PruneEmptyRulesTest.xml:1]: @snuyanzin PruneEmptyRulesTest

Comment (liuyongvs) [reply flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PruneEmptyRulesTest.xml:1]: if you think it should remove the test. i think we also should remove JoinPushExpressionsRuleTest this test?
what do you think?

**Comment by liuyongvs** [comment]
hi @snuyanzin will you look again?

**Comment by liuyongvs** [comment]
Hi @snuyanzin will you look again?

File: flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PruneEmptyRulesTest.xml
Comment (liuyongvs) [reply flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PruneEmptyRulesTest.xml:1]: @snuyanzin do same with FlinkLimit0RemoveRule now

Former test for [[FlinkPruneEmptyRules]] which now replaced by Calcite's
 * [[PruneEmptyRules.JOIN_RIGHT_INSTANCE]].

**Comment by liuyongvs** [comment]
hi @snuyanzin will you look again?

**Comment by liuyongvs** [comment]
hi @snuyanzin will you look again?

**Review by snuyanzin — APPROVED** [review]

**Pull request merged**